packages/flutter_tools/lib/src/version.dart

```diff
@@ -275,6 +292,17 @@ class FlutterVersion {
 ''';
   }
 
+  @visibleForTesting
+  static String newVersionAvailableMessage() {
+    return '''
+  ╔════════════════════════════════════════════════════════════════════════════╗
+  ║ A new version of Flutter is available!                                     ║
+  ║                                                                            ║
+  ║ To update to the latest version, run "flutter upgrade".                    ║
+  ╚════════════════════════════════════════════════════════════════════════════╝
+''';
```

Comment:
it'd be great if we could say what the current version was, what the new version was, and what channel the user was on.

Reply:
I had a go at this, but it turned into more work than I expected - we now carry around and cache both a remote commit date and also the version. I've pushed it to a branch for now so you can have a quick scan in case it's going in the wrong direction or you'd just rather it another PR for review.

https://github.com/DanTup/flutter/commit/d474a5499e6ae8a6878f92f203df6e11ab61a1d3

The tests are all passing, but there's a lot here and it's slightly complicated to manually test (without rigging code or resetting the git pointer to an old revision and just running with the changes in working dir) so it's a little risky (which I guess goes for this PR too!).

Reply:
doing it in another PR (or not at all for now, if it's not tractable) is fine.